cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackendImpl.java

```diff
@@ -204,21 +204,42 @@ public void createCollection(String dbName, String collectionName, long collecti
                 default:
              }
             options = new IndexOptions().name("cyg_raw_opt");
-            db.getCollection(collectionName).createIndex(keys, options);
+            createIndex(db, collectionName, keys, options);
         } catch (Exception e) {
             LOGGER.warn("Error in collection " + collectionName + " creating index ex=" + e.getMessage());
         } // try catch
         try {
             if (dataExpiration != 0) {
                 keys = new BasicDBObject().append("recvTime", 1);
                 options = new IndexOptions().name("cyg_raw_exp").expireAfter(dataExpiration, TimeUnit.SECONDS);
-                db.getCollection(collectionName).createIndex(keys, options);
+                createIndex(db, collectionName, keys, options);
```

Comment:
Should the try/catch be removed (given that createIndex() will deal with the exception internally)?

Reply:
Ok, will remove try/catch.

Reply:
Fixed in b6045a0